File: 05-AI销售培训系统/backend/app/schemas/training.py
"""Pydantic schemas for training sessions and rounds."""
from datetime import datetime
from pydantic import BaseModel, Field


class TrainingSessionCreate(BaseModel):
    """Create a new training session."""
    customer_type: str = Field(..., pattern="^(picky|price|hesitant|expert)$")
    product_context: str | None = Field(None, max_length=2000)


class TrainingSessionResponse(BaseModel):
    """Training session in list/detail views."""
    id: str
    user_id: str
    customer_type: str
    product_context: str | None = None
    overall_score: float | None = None
    total_rounds: int
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TrainingRoundResponse(BaseModel):
    """A single round in a training session."""
    id: str
    training_session_id: str
    round_number: int
    user_response: str
    customer_response: str
    coach_hint: str | None = None
    scores: dict | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class TrainingRespondRequest(BaseModel):
    """Send a sales response in the training session."""
    response: str = Field(..., min_length=1, max_length=3000)


class TrainingReportResponse(BaseModel):
    """Complete training report."""
    session: TrainingSessionResponse
    rounds: list[TrainingRoundResponse]
    score_trend: list[dict]  # [{round, overall, fluency, persuasiveness, ...}]
    strengths: list[str]
    improvements: list[str]
    recommendation: str

File: 05-AI销售培训系统/frontend/src/api/training.ts
import apiClient from './client'

export interface TrainingSession {
  id: string
  user_id: string
  customer_type: string
  product_context: string | null
  overall_score: number | null
  total_rounds: number
  status: string
  created_at: string
  updated_at: string
}

export interface TrainingRound {
  id: string
  training_session_id: string
  round_number: number
  user_response: string
  customer_response: string
  coach_hint: string | null
  scores: Record<string, number> | null
  created_at: string
}

export interface CustomerType {
  key: string
  name: string
  icon: string
  difficulty: number
  persona: string
}

export interface ScoreTrendEntry {
  round: number
  overall: number
  [dimension: string]: number
}

export interface TrainingReport {
  session: TrainingSession
  rounds: TrainingRound[]
  score_trend: ScoreTrendEntry[]
  strengths: string[]
  improvements: string[]
  recommendation: string
}

export const trainingApi = {
  // Sessions
  listSessions(page = 1, pageSize = 20) {
    return apiClient.get<TrainingSession[]>('/training/sessions', {
      params: { page, page_size: pageSize },
    })
  },

  createSession(customerType: string, productContext?: string) {
    return apiClient.post<TrainingSession>('/training/sessions', {
      customer_type: customerType,
      product_context: productContext,
    })
  },

  getSession(sessionId: string) {
    return apiClient.get<TrainingSession>(`/training/sessions/${sessionId}`)
  },

  deleteSession(sessionId: string) {
    return apiClient.delete(`/training/sessions/${sessionId}`)
  },

  // Rounds
  getRounds(sessionId: string) {
    return apiClient.get<TrainingRound[]>(`/training/sessions/${sessionId}/rounds`)
  },

  // SSE streaming respond
  respond(sessionId: string, response: string, token: string): Promise<Response> {
    return fetch(`/api/training/sessions/${sessionId}/respond`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify({ response }),
    })
  },

  // End & Report
  endSession(sessionId: string) {
    return apiClient.post(`/training/sessions/${sessionId}/end`)
  },

  getReport(sessionId: string) {
    return apiClient.get<TrainingReport>(`/training/sessions/${sessionId}/report`)
  },

  // Customer types
  getCustomerTypes() {
    return apiClient.get<CustomerType[]>('/training/customer-types')
  },
}
